test/error.jl

```diff
@@ -99,3 +99,27 @@ end
         @test s == "MethodError: no method matching f44319(::Int$(Sys.WORD_SIZE))\n\nClosest candidates are:\n  f44319()\n   @ $curmod_str none:0\n"
     end
 end
+
+@testset "All types ending with Exception or Error subtype Exception" begin
+    function test_exceptions(mod, visited=Set{Module}(), mod_exp=Symbol(mod))
+        if mod ∉ visited
+            push!(visited, mod)
+            for name in names(mod, all=true)
+                isdefined(mod, name) || continue
+                joined_expr = Expr(:., mod_exp, QuoteNode(name))
+                value = eval(joined_expr)
```

Comment:
Why not simply `value = getproperty(mod, name)`?

Reply:
Like most uses of `eval`, because the author (me) wasn't aware of the better method. Thanks!